### PR TITLE
add HTTP parser benchmarks

### PR DESCRIPTION
**Pull request by mfelsche**
I had to make the `_HTTPParser._restart` method public, to get the benchmark set up.

### CODE REVIEW

**Comment by jemc** [comment]
Instead of adding `ponybench` as a dependency of the `http` library, why not make `_HTTPParser` a public class instead?

**Comment by mfelsche** [comment]
@jemc ... and moving the parser benchmarks somewhere else? I don't see the benefits tbh and i don't mind adding a dependency from stdlib. Change my mind. ;)

File: http/_bench.pony [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+use "buffered"
+use "debug"
+
+class PrivateBenchmarks is BenchmarkList
+  fun tag benchmarks(bench: PonyBench tag) =>
+    Debug("benching private benchmarks")
+    bench(_SimpleGetRequestBenchmark)
+    bench(_FormSubmissionRequestBenchmark)
+    bench(_SplitFormSubmissionRequestBenchmark)
+//    bench(_MultipartFileUploadBenchmark)
```
Comment (StephenStrickland) [inline http/_bench.pony]: Is there a reason why you are committing commented out code?

**Comment by mfelsche** [comment]
Actually, @jemc, having thought about it, i agree with you, making the parser public and avoid having ponybench in the http sources.

File: http/_bench.pony (hunk 1 above)
Comment (mfelsche) [reply http/_bench.pony]: I just wanted to share the basic setup already. Those benchmarks are going to be implemented as part of this PR. If i eouldnt comment them, they would timeout or run very very long, as they do not complete.

Comment (mfelsche) [reply http/_bench.pony]: i added the benchmarks that were previously commented out.

**Comment by mfelsche** [comment]
@jemc we cool here?

**Pull request merged**